src/decorator/columns/SoftDeleteDateColumn.ts

```diff
@@ -0,0 +1,22 @@
+import {ColumnOptions} from "../options/ColumnOptions";
+import {ColumnMetadataArgs} from "../../metadata-args/ColumnMetadataArgs";
+import {getMetadataArgsStorage} from "../../index";
+
+/**
+ * This column will store an soft delete date of the deleted object.
+ * This date is set when you delete the object, and set to null when you restore it
+ */
+export function SoftDeleteDateColumn(options?: ColumnOptions): Function {
```

Comment:
What if people want to use a flag instead of date? Something like `isRemoved: boolean`. I think we need to support that as well.

Reply:
If we'll do so we can simply call it `SoftDeleteColumn`

Reply:
I have always seen it as a date. I suppose we could support both options, but I think that complicates things.

TBH, I would not use it if it was only a flag. In every single project I have, I need to know when something was deleted.

Reply:
I had several projects where we used just a flag. And when things are getting really complex and we need a deletion date we store in a separate table where all *activity* on the given object is stored.

Reply:
Would be great if you can implement this functionality. If we'll do do it later, but we definitely need to rename decorator right now.

Reply:
I would argue that having it as a date has a more general scope of usage than it being only a flag:

 * it gives information on when the entry was deleted (useful in debugging)
 * a nice side effect is that you can expire an entry by setting delete time in the future

Unfortunately I don't have knowledge if having a boolean value over date is better performance wise so I can't argue on that point.

As a last point, library (in general terms) should provide feature that covers as much use cases as possible for the feature in question.